setup.sh

```diff
@@ -1,19 +1,50 @@
 #!/bin/bash
+set -x
 
+# This script installs juju version 2 and all the dependencies.
+
+# Refresh the potentially stale apt cache.
 apt-get update -qq
+# Install software tools such as add-apt-repository.
 apt-get install -qy software-properties-common
+# Install the ppa for Juju and charm tools.
 apt-add-repository -y ppa:juju/stable
```

Comment:
If you add -u to this command, it will fetch the index from the PPA directly after adding, and you can nix the apt-get update -qq below. This was a protip provided by @jrwren